### PR TITLE
fix: Coerce historyFallback to a boolean

### LINKED ISSUE #47: Fix webpack-plugin-serve historyFallback type
---

PR #48

---

**About**

There's a regression in `historyFallback` type of webpack-plugin-serve in 1.2.0 (https://github.com/shellscape/webpack-plugin-serve/issues/190) due to which **webpack-config** 4.2.0 fails to work.

**Possible implementation**

To solve, coerce the value to a boolean.

---

**Analysis**
_Should be filled out in the Analysis phase. Needs to be approved by another engineer before implementation starts. See [How to write a good analysis](http://handbook.adverity.net/kanban/#how-to-write-an-analysis-of-an-issue)_

---

**Tasks**
_A very detailed and technical list of tasks. Should be filled out after Analysis._

- [ ] task 1
- [ ] ..

### PR DESCRIPTION
**Pull request by bebraw**
---
Closes #47
---

**How to test (feature)**

There's a new `dev` script behind `demo` now. Run `npm run dev` within there to see webpack-plugin-serve working.

**How to test (potential regressions)**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -411,7 +411,7 @@ function dontParse(paths: webpack.Module["noParse"]): webpack.Configuration {
 // to `staticPaths`.
 function webpackPluginServe({
   staticPaths,
-  historyApiFallback: historyFallback,
+  historyApiFallback,
```
Comment (bebraw) [inline src/index.ts:414]: This will be `undefined` by default and that was the issue as webpack-plugin-serve has a stricter type validation now.

**Review by TkDodo — APPROVED** [review]

**Comment by TkDodo** [comment]
hmm, tests are now failing with:

```
> @pmedianetwork/webpack-config@4.2.0 test /home/ubuntu/jenkins-slave/workspace/webpack-config/pull-request-test
> jest test.js

Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
FAIL ./test.js
  ● Test suite failed to run

    Expected a value of type `boolean | Type<{}>` for `historyFallback` but received `undefined`.

      354 |     return {
      355 |         plugins: [
    > 356 |             new webpack_plugin_serve_1.WebpackPluginServe(tslib_1.__assign({ hmr: "refresh-on-failure", progress: "minimal", historyFallback: historyFallback, middleware: function (app) {
          |             ^
      357 |                     return app.use(function (ctx, next) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
      358 |                         return tslib_1.__generator(this, function (_a) {
      359 |                             switch (_a.label) {

      at new StructError (node_modules/superstruct/src/struct.ts:71:18)
      at validate (node_modules/superstruct/src/struct.ts:170:19)
      at validate (node_modules/webpack-plugin-serve/lib/validate.js:63:28)
      at new WebpackPluginServe (node_modules/webpack-plugin-serve/lib/index.js:58:19)
      at webpackPluginServe (dist/index.js:356:13)
      at Object.<anonymous> (demo/webpack.config.js:8:27)
      at Object.<anonymous> (test.js:3:16)

  console.error
    ⬢ wps: An option was passed to WebpackPluginServe that is not valid

      354 |     return {
      355 |         plugins: [
    > 356 |             new webpack_plugin_serve_1.WebpackPluginServe(tslib_1.__assign({ hmr: "refresh-on-failure", progress: "minimal", historyFallback: historyFallback, middleware: function (app) {
          |             ^
      357 |                     return app.use(function (ctx, next) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
      358 |                         return tslib_1.__generator(this, function (_a) {
      359 |                             switch (_a.label) {

      at forceError (node_modules/webpack-plugin-serve/lib/log.js:26:3)
      at new WebpackPluginServe (node_modules/webpack-plugin-serve/lib/index.js:61:7)
      at webpackPluginServe (dist/index.js:356:13)
      at Object.<anonymous> (demo/webpack.config.js:8:27)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        2.004 s
```

**Comment by bebraw** [comment]
@TkDodo Ah, I see. Let me check that.

**Comment by bebraw** [comment]
@TkDodo I fixed the issue - the thing is that we have to build the project before testing so now it's doing that.

**Pull request merged**